ledger/txtail.go

```diff
@@ -101,32 +119,31 @@ func (t *txTail) committedUpTo(rnd basics.Round) basics.Round {
 			delete(t.recent, r)
 		}
 	}
+	for lv := range t.lastValid {
+		if lv < rnd {
+			delete(t.lastValid, lv)
+		}
+	}
 
 	return (rnd + 1).SubSaturate(maxlife)
 }
 
 func (t *txTail) isDup(proto config.ConsensusParams, current basics.Round, firstValid basics.Round, lastValid basics.Round, txid transactions.Txid, txl txlease) (bool, error) {
```

Comment:
I think this optimization makes sense, but I think that this function signature is a bit dangerous now. This will fail to find a duplicate `txid` if the `lastValid` argument isn't exactly equal to the `LastValid` of the transaction. I think that `isDup` should instead take in a proto and a `transactions.Transaction` to avoid this situation.

Reply:
I think it would also be ok to just change the argument names to `txFirstValid` and `txLastValid`.

Reply:
Good idea--I think I'll just have the entire transaction passed in.

Reply:
I can fix the interface in a separate commit--will leave it as-is for now.